backend/utilities/helpers/EnvHelper.py

```diff
@@ -39,12 +50,13 @@ def __init__(self, **kwargs) -> None:
         self.AZURE_OPENAI_API_VERSION = os.getenv('AZURE_OPENAI_API_VERSION', '')
         self.AZURE_OPENAI_STREAM = os.getenv('AZURE_OPENAI_STREAM', '')
         self.AZURE_OPENAI_EMBEDDING_MODEL = os.getenv('AZURE_OPENAI_EMBEDDING_MODEL', '')
+        self.AZURE_TOKEN_PROVIDER = get_bearer_token_provider(DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default")
          # Set env for OpenAI SDK
-        self.OPENAI_API_TYPE = "azure"
+        self.OPENAI_API_TYPE = "azuread" if self.AUTH_TYPE == 'rbac' else "azure"
         self.OPENAI_API_BASE = f"https://{os.getenv('AZURE_OPENAI_RESOURCE')}.openai.azure.com/"
         self.OPENAI_API_KEY = self.AZURE_OPENAI_KEY
         self.OPENAI_API_VERSION = self.AZURE_OPENAI_API_VERSION
-        os.environ["OPENAI_API_TYPE"] = "azure"
+        os.environ["OPENAI_API_TYPE"] = "azuread" if self.AUTH_TYPE == 'rbac' else "azure"
```

Comment:
These two conditions look like they're repeated, can you extract those into a variable?